R/pp_check.R

```diff
@@ -175,14 +172,38 @@ pp_check.brmsfit <- function(object, type, ndraws = NULL, nsamples = NULL,
       do_call(compute_loo, c(pred_args, criterion = "psis"))
     )
   }
-  if (!is.null(group)) {
-    ppc_args$group <- data[[group]]
-  }
-  if (!is.null(x)) {
-    ppc_args$x <- data[[x]]
-    if (!is_like_factor(ppc_args$x)) {
-      ppc_args$x <- as.numeric(ppc_args$x)
+  
+  # censored responses are misleading when displayed in pp_check
+  bterms <- brmsterms(object$formula)
+  cens <- get_cens(bterms, data, resp = resp)
+  if (!is.null(cens) & type != 'km_overlay') {
+    warning2("Censored responses are not shown in 'pp_check'.")
+    take <- !cens
+    if (!any(take)) {
+      stop2("No non-censored responses found.")
+    }
+    ppc_args$y <- ppc_args$y[take]
+    ppc_args$yrep <- ppc_args$yrep[, take, drop = FALSE]
+    if (!is.null(ppc_args$group)) {
+      ppc_args$group <- ppc_args$group[take]
+    }
+    if (!is.null(ppc_args$x)) {
+      ppc_args$x <- ppc_args$x[take]
+    }
+    if (!is.null(ppc_args$psis_object)) {
+      # tidier to re-compute with subset
+      psis_args <- c(pred_args, criterion = "psis")
+      psis_args$newdata <- kidney[take, ]
```

Comment:
I think using the kidney data here is a typo, right?

Reply:
Ahh sorry, thanks for spotting! Have fixed